modules/apps/portal-security/portal-security-permission-impl/src/main/java/com/liferay/portal/security/permission/internal/factory/PermissionCheckerFactoryImpl.java

```diff
@@ -62,12 +48,27 @@ public PermissionChecker create(User user) {
 		return permissionChecker;
 	}
 
-	public void destroy() {
+	@Activate
+	protected void activate(BundleContext bundleContext) throws Exception {
+		Class<PermissionChecker> clazz =
```

Comment:
@W2jx
As you are moving PermissionCheckerFactoryImpl out, the class loader used to load PropsValues.PERMISSIONS_CHECKER is actually changed.

I am curious why this works as before your change, PermissionCheckerFactoryImpl is in portal-impl, it is using portal class loader, after your change, it will use this bundle portal-security-permission-impl class loader.

While the default value for PropsValues.PERMISSIONS_CHECKER is "com.liferay.portal.security.permission.AdvancedPermissionChecker" which stays in portal-impl.

How could a bundle class loader see it? unless it is exported from portal-impl.

Could you please help me to verify by debugging?

One of the reason we move things out into modules is to have clear boundary for code, thus depending on a hidden dependency like this is not a clean solution. The clean solution would be also moving AdvancedPermissionChecker out and remove the portal property "permissions.checker" as we have provided osgi way to extend the permission checker.